metafields.go

```diff
@@ -0,0 +1,196 @@
+package shopify
+
+import (
+	"strconv"
+	"time"
+)
+
+// Metafields are a flexible way to attach additional information to a Shopify resource (e.g. Product, Collection, etc.).
+type Metafields []Metafield
+
+// GetByKey gets a metafield via its key
+func (m Metafields) GetByKey(key string) (interface{}, error) {
+	var err error
+	for _, metafield := range m {
+		if metafield.Key == key {
+			switch metafield.Type {
+			case NumberIntegerMetaFieldType:
+				var value int64
+				if metafield.Value != "" {
```

Comment:
How come you are checking to see if an empty string here?

Reply:
Will test this.

Reply:
So this is happening because it errors trying to parseInt an empty string.